resources/api-gateway/templates/deployment.yaml

```diff
@@ -39,6 +39,7 @@ spec:
             - --generated-objects-labels={{ .Values.config.generatedObjects.labels }}
             {{- with .Values.global.ingress.domainName }}
             - --domain-whitelist={{ . | join ", " }}
+            - --default-domain-name={{ . | join ", " }}
```

Comment:
I think we can refactor this a bit. 
Currently in `values.yaml` we have a `domainWhiteList` field that is not used, as we only use the value from `global.ingress.domainName`. 

```yaml
config:
  domainWhiteList:
    - kyma.local
```

Imho we can improve this by defining 2 helm functions:
- getDomainWhiteList: which does "`domainWhiteList + global.ingress.domainName | uniqe`" 
- getDefaultDomain: which uses a new `defaultDomain` variable if set, and default to `global.ingress.domainName` if empty

Reply:
done like suggested